tensorboard/webapp/plugins/npmi/util/filter_annotations_test.ts

```diff
@@ -0,0 +1,455 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {
+  AnnotationDataListing,
+  ArithmeticElement,
+  ArithmeticKind,
+  Operator,
+  MetricFilterListing,
+} from '../store/npmi_types';
+import {filterAnnotations, removeHiddenAnnotations} from './filter_annotations';
+
+describe('filter annotations utils', () => {
+  const annotationData: AnnotationDataListing = {
```

Comment:
General feedback: I think you are repeating this type of fixture creation in many tests.
For ease of refactor when the data structure changes later, it is often a good idea to use a builder/factory method.

Also, when reading the test specs, it is ideal to be able to tell why you get the result you are asserting without jumping too much around the file (i.e., metricFilters is used over and over and I have to remind myself how it is defined). 

If you think any of this is a good idea, feel free to refactor away here or in a follow up.

Reply:
I think this is a good point. I guess a follow-up PR would make sense here.

Reply:
Did that now for the annotation data.